### PR TITLE
Add tests for grids/LayoutGrids.html

### PR DESCRIPTION
**Pull request by spectranaut**
This replace PR: https://github.com/w3c/aria-practices/pull/755

I addressed all your feedback here, @tatermelon ! The commits are in the same order, so you reviewed up to the "Satisfy linter" commit.

### CODE REVIEW

**Comment by mcking65** [comment]
@jessebeach, @tatermelon, @sh0ji, this PR is ready for your review.

File: test/tests/LayoutGrids.js [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+
+const findFocusable = function (/* selector */) {
+  const selector = arguments[0];
+  const original = document.activeElement;
+  const focusable = Array.from(document.querySelectorAll(selector))
+    .map((parent) => {
+      const candidates = [parent, ...parent.querySelectorAll('*')];
+      for (let candidate of candidates) {
+
+        candidate.focus();
```
[the remark this answers is not in the document]
Comment (sh0ji) [reply test/tests/LayoutGrids.js]: I generally think this is okay, but we will at least want to validate it in the testing environment (see my comment on 56). The alternative might be to keep a list of focusable selectors, but that quickly gets out of hand when you consider things like `button:not(disabled):not([tabindex="-1"])`.

File: test/tests/LayoutGrids.js [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+  const el = await session.findElement(By.css(selector));
+  await el.click();
+
+  if (await el.isEnabled()) {
+    return clickUntilDisabled(session, selector);
+  }
+};
+
+const checkActiveElement = function (/* gridcellsSelector, index */) {
+  let gridcellsSelector = arguments[0];
```
[the remark this answers is not in the document]
Comment (spectranaut) [reply test/tests/LayoutGrids.js]: This function is passed to webdriver.executeScript (ctrl-f for executeScript [here](http://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_WebDriver.html) -- sorry the docs aren't great for linking). It's a requirement for sending js with arguments to execute in the browser.

File: test/tests/LayoutGrids.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (spectranaut) [reply test/tests/LayoutGrids.js]: I think @sh0ji is right here. I think the direction we want to take is to encode a list of focusable selectors. This matches the direction we got from @mcking65 last week as well.

The purpose here was to mutate the doc, actually, @jessebeach -- layout grid has some complicate logic for which element in a gridcell will take focus and, therefore, except keyboard interactions. The point of this function is to find AND focus on the focusable element in the gridcell. Also re: comment on 45, yup -- focus will not move to elements that are not focusable in these cases.

I'll switch this complicated logic over to a list of focusable elements based on the specific examples. I think it will be much clearer to read and understand and matches the direction we are going with other tests. Can I make a subtask for this and get to it this week -- and merge this PR for now, to unblock the landing of the util function refactor #787 which changes some code in this PR?

File: test/tests/LayoutGrids.js [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+  return focusable;
+};
+
+const focusWithin = function (/* element */) {
+  // Assumption: `element is` focusable or contains only one focusable element.
+  let element = arguments[0];
+  let candidates = [element, ...element.querySelectorAll('*')];
+
+  for (let candidate of candidates) {
+    candidate.focus();
```
[the remark this answers is not in the document]
Comment (spectranaut) [reply test/tests/LayoutGrids.js]: oh thanks for being concerned @jessebeach and for looking this up @sh0ji -- interesting edge cases. based on experience I assumed focus would only work on focusable elements.

File: test/tests/LayoutGrids.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sh0ji) [reply test/tests/LayoutGrids.js]: Sure, I'll merge this now and look for the update later this week.

**Comment by sh0ji** [comment]
Merging this to unblock #787. Per https://github.com/w3c/aria-practices/pull/784#discussion_r204474915, the focusable elements will be targeted via a list of selectors in a future PR.

File: test/tests/LayoutGrids.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jessebeach) [reply test/tests/LayoutGrids.js]: >  It's a requirement for sending js with arguments to execute in the browser.

huh, I had no idea. Good to know. Thank you for the explanation!

File: test/tests/LayoutGrids.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jessebeach) [reply test/tests/LayoutGrids.js]: > The purpose here was to mutate the doc, actually, @jessebeach

Ok, good to know. Thank you for the explanation.
